pkg/types/baremetal/defaults/platform.go

```diff
@@ -69,4 +96,59 @@ func SetPlatformDefaults(p *baremetal.Platform, c *types.InstallConfig) {
 			p.IngressVIP = vip[0]
 		}
 	}
+
+	if p.ProvisioningInterface == "" {
+		p.ProvisioningInterface = ProvisioningInterface
+	}
+
+	if p.ProvisioningNetworkCIDR == "" {
+		p.ProvisioningNetworkCIDR = ProvisioningNetworkCIDR
+	}
+
+	if p.ProvisioningHTTPPort == "" {
+		p.ProvisioningHTTPPort = ProvisioningHTTPPort
+	}
+
+	_, provNet, _ := net.ParseCIDR(p.ProvisioningNetworkCIDR)
+
+	if p.ProvisioningIP == "" {
+		ip, err := cidr.Host(provNet, 3)
+		if err == nil {
+			p.ProvisioningIP = ip.String()
+		}
+	}
+
+	if p.ProvisioningDHCPStart == "" {
+		ip, err := cidr.Host(provNet, 20)
+		if err == nil {
+			p.ProvisioningDHCPStart = ip.String()
+		}
+	}
+
+	if p.ProvisioningDHCPEnd == "" {
+		ip, err := cidr.Host(provNet, 200)
+		if err == nil {
+			p.ProvisioningDHCPEnd = ip.String()
+		}
+	}
+
+	if p.DeployKernelURL == "" {
+		p.DeployKernelURL = strings.Replace(DeployKernelURL, "#ProvisioningIP#", p.ProvisioningIP, 1)
+	}
+
+	if p.DeployRamdiskURL == "" {
+		p.DeployRamdiskURL = strings.Replace(DeployRamdiskURL, "#ProvisioningIP#", p.ProvisioningIP, 1)
+	}
+
+	if p.IronicEndpoint == "" {
+		p.IronicEndpoint = strings.Replace(IronicEndpoint, "#ProvisioningIP#", p.ProvisioningIP, 1)
+	}
+
+	if p.IronicInspectorEndpoint == "" {
+		p.IronicInspectorEndpoint = strings.Replace(IronicInspectorEndpoint, "#ProvisioningIP#", p.ProvisioningIP, 1)
+	}
```

Comment:
Could we pass PROVISIONING_IP optionally to the BMO, then copy/paste this code to e.g https://github.com/metal3-io/baremetal-operator/blob/master/pkg/provisioner/ironic/ironic.go#L51 such that we don't need the URL/Endpoint values exposed by the installer?

Reply:
We certainly can.. everyone seems to want to do this in a different place.  I'd like to chat about this as a group and see if we can come to a conclusion on configuration items and where values will be derived.

Reply:
For reference this discussion happened, and the consensus was to calculate these values in the MAO, then we avoid exposing the internal details of how we need the BMO configured to the user, but also avoid hard-coding assumptions about that (somewhat openshift specific) BMO configuration inside the BMO.  cc @dhellmann @sadasu